src/lex/mod.rs

```diff
@@ -191,22 +191,29 @@ impl Lexer {
     ///
     /// Before: b"    // some comment\n /*multi comment*/hello   "
     /// After:  b"hello   "
-    fn consume_whitespace(&mut self) {
+    fn consume_whitespace(&mut self) -> String {
         // there may be comments following whitespace
+        let mut whitespace = String::new();
         loop {
             // whitespace
             while self.peek().map_or(false, |c| c.is_ascii_whitespace()) {
-                self.next_char();
+                if let Some(c) = self.next_char() {
+                    whitespace.push(c.into());
```

Comment:
This might now be better as a match + loop so you don't have to check the `Some` condition twice. Up to you though.

Reply:
Not sure what you mean by that, but this doesn't look too bad to me.